tensorflow/python/kernel_tests/conv_ops_3d_test.py

```diff
@@ -223,7 +224,10 @@ def testConv3D1x1x1Filter(self):
         expected=expected_output)
 
   def testConv3D1x1x1Filter2x1x1Dilation(self):
-    if test.is_gpu_available(cuda_only=True) or test_util.IsMklEnabled():
+    ctx = context.context()
+    is_eager = ctx is not None and ctx.executing_eagerly()
+    if test.is_gpu_available(cuda_only=True) or \
+      test_util.IsMklEnabled() and is_eager is False:
```

Comment:
Let's add a parenthesis to make the execution order clear. It will help us remove `\` at the end.
```Python
if (test.is_gpu_available(cuda_only=True) or 
  test_util.IsMklEnabled()) and is_eager is False:
```
Alternatively, we can define another variable `is_gpu_available_or_mkl_enabled` as well.

Reply:
Since `and` has a higher precedence than 'or', I think existing code already represent the correct order:
<pre>
if test.is_gpu_available(cuda_only=True or 
  (test_util.IsMklEnabled()) and is_eager is False):
</pre>

Which means the case is only enabled when:
1. GPU, either eager or legacy mode;
2. MKL CPU, but only for legacy mode.

Reply:
Got it. Could you please add the parentheses as you did in the code snippet? It doesn't hurt to make the order clear to the readers.

Reply:
Fair enough. Parenthesis added now.